### PR TITLE
8345353: Test for JDK-8344800 W3C DTDs and XSDs in the built-in Catalog

### PR DESCRIPTION
**Pull request by JoeWang-Java**
Add test for JDK-8344800 W3C DTDs and XSDs in the built-in Catalog

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8345353](https://bugs.openjdk.org/browse/JDK-8345353): Test for JDK-8344800 W3C DTDs and XSDs in the built-in Catalog (**Bug** - P4)

### Reviewers
 * [Lance Andersen](https://openjdk.org/census#lancea) (@LanceAndersen - **Reviewer**) Review applies to [f6f8f0f6](https://git.openjdk.org/jdk/pull/22506/files/f6f8f0f6b5047c6c88e6f802a6cc240000330391)
 * [Roger Riggs](https://openjdk.org/census#rriggs) (@RogerRiggs - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/22506/head:pull/22506` \
`$ git checkout pull/22506`

Update a local copy of the PR: \
`$ git checkout pull/22506` \
`$ git pull https://git.openjdk.org/jdk.git pull/22506/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 22506`

View PR using the GUI difftool: \
`$ git pr show -t 22506`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/22506.diff">https://git.openjdk.org/jdk/pull/22506.diff</a>

</details>
<details><summary>Using Webrev</summary>

[Link to Webrev Comment](https://git.openjdk.org/jdk/pull/22506#issuecomment-2513408799)
</details>

### CODE REVIEW

**Review by LanceAndersen — APPROVED** [review]
Looks OK.  As this is a new test, I would have preferred this to be a junit test, but lets get this back given where we are for 24 and can address afterwards

File: test/jaxp/javax/xml/jaxp/unittest/common/jdkcatalog/JDKCatalogTest.java [hunk 1]
```diff
@@ -0,0 +1,275 @@
...
+    private String generateXMLWithDTDRef(String publicId, String systemId)
+            throws Exception {
+        Path path = Paths.get(SRC_DIR + "/dtdtest.xml");
+        String xmlString = Files.lines(path).map(line -> {
+            if (line.contains(PUBLIC_ID)) {
+                line = line.replace(PUBLIC_ID, publicId);
+            }
+            if (line.contains(SYSTEM_ID)) {
+                line = line.replace(SYSTEM_ID, systemId);
+            }
```
Comment (RogerRiggs) [inline test/jaxp/javax/xml/jaxp/unittest/common/jdkcatalog/JDKCatalogTest.java:217]: The `if ... contains ...` isn't necessary; String.replace is a no-op if the string is not present.

**Review by RogerRiggs — APPROVED** [review]

**Review by RogerRiggs — APPROVED** [review]
Looks good

**Pull request closed without merging**